OutOfSchool/OutOfSchool.EmailSender/EmailSenderService.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+using OutOfSchool.Services.Models;
+using OutOfSchool.Services.Repository;
+
+using SendGrid;
+using SendGrid.Helpers.Mail;
+
+namespace OutOfSchool.EmailSender;
+
+public class EmailSenderService : IEmailSenderService
+{
+    private readonly IOptions<EmailOptions> emailOptions;
+    private readonly ISendGridClient sendGridClient;
+    private readonly ILogger<EmailSenderService> logger;
+    private readonly IEmailOutboxRepository emailOutboxRepository;
+
+    public EmailSenderService(
+        IOptions<EmailOptions> emailOptions,
+        ISendGridClient sendGridClient,
+        ILogger<EmailSenderService> logger,
+        IEmailOutboxRepository emailOutboxRepository)
+    {
+        this.emailOptions = emailOptions;
+        this.sendGridClient = sendGridClient;
+        this.logger = logger;
+        this.emailOutboxRepository = emailOutboxRepository;
+    }
+
+    public async Task SendAsync(string email, string subject, (string html, string plain) content)
+    {
+        var outboxMessage = new EmailOutbox()
+        {
+            Email = email,
+            Subject = subject,
+            HtmlContent = content.html,
+            PlainContent = content.plain,
+            CreationTime = DateTime.Now,
+        };
+        await emailOutboxRepository.Create(outboxMessage);
```

Comment:
i think it's better to add a `Base64` encoding step for both html & plain content before saving to DB. As those emails can have special symbols or contain keywords that may break database when exporting/importing or backing up.